clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -643,7 +643,7 @@ public void handle(SyncGroupResponse syncResponse,
                 } else if (error == Errors.UNKNOWN_MEMBER_ID
                         || error == Errors.ILLEGAL_GENERATION) {
                     log.debug("SyncGroup failed: {}", error.message());
-                    resetGeneration();
+                    resetGeneration(this.getClass().getName() + " encountering " + error);
```

Comment:
It's not necessary to pass in the class name, because log4j should be able to reveal it. Also, we are double logging here, conflicting with L525 and L645. Consider removing some of the loggings

Reply:
@guozhangwang